src/main/kotlin/app/revanced/patches/twitch/ad/video/fingerprints/CheckAdEligibilityLambdaFingerprint.kt

```diff
@@ -0,0 +1,27 @@
+package app.revanced.patches.twitch.ad.video.fingerprints
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.extensions.or
+
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import app.revanced.patches.twitch.ad.video.annotations.VideoAdsCompatibility
+import org.jf.dexlib2.AccessFlags
+
+@Name("check-ad-eligibility-fingerprint")
+
+@VideoAdsCompatibility
+@Version("0.0.1")
+object CheckAdEligibilityLambdaFingerprint : MethodFingerprint(
+    "Lio/reactivex/SingleSource;",
+    AccessFlags.PRIVATE or AccessFlags.FINAL or AccessFlags.STATIC,
+    listOf(
+        "Ltv/twitch/android/shared/ads/models/AdRequestInfo;",
+        "Ltv/twitch/android/shared/ads/eligibility/AdEligibilityFetcher;",
+        "Ltv/twitch/android/util/Optional;"
+    ),
```

Comment:
The full descriptor of types are not checked. This fingerprint works equivalently with the following suggestion:
```suggestion
    "L",
    AccessFlags.PRIVATE or AccessFlags.FINAL or AccessFlags.STATIC,
    listOf("L", "L", "L"),
```